lib/galaxy/datatypes/binary.py

```diff
@@ -291,9 +291,14 @@ def get_chunk(self, trans, dataset, offset=0, ck_size=None):
                             # Below code will remove spaces between each tag.
                             bamline_modified = ('\t').join(bamline.split()[:11] + [(' ').join(bamline.split()[11:])])
                             ck_data = "%s\n%s" % (ck_data, bamline_modified)
+                    else:
+                        # Nothing to enumerate; we've either offset to the end
+                        # of the bamfile, or there is no data. (possible with
+                        # header-only bams)
+                        offset = -1
             except Exception as e:
                 offset = -1
-                ck_data = "Could not display BAM file, error was:\n%s" % e.message
+                ck_data = "Could not display BAM file, error was:\n%s" % str(e)
```

Comment:
The `str()` casting shouldn't be necessary, I think, but we have tons of these anyway.